### ISSUE TITLE
Crawler stuck without exiting

### ISSUE THREAD

**Issue by rgaudin** [webrecorder/browsertrix-crawler#371]
Here's the tail of a crawl attempt at `http://www.toratemetfreeware.com/online/`.
It seems to be hitting a lot of navigation timeout.

This is using the latest crawler (`0.10.4`).

Issue is that the crawler apparently stopped producing output at 2023-09-10T11:34. It had not exited the next day at 8am.

```
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.301Z","context":"behavior","message":"Running behaviors","details":{"frames":1,"frameUrls":["http://www.toratemetfreeware.com/online/f_00847.html"],"page":"http://www.toratemetfreeware.com/online/f_00847.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.301Z","context":"behavior","message":"Run Script Started","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00847.html","page":"http://www.toratemetfreeware.com/online/f_00847.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.306Z","context":"behavior","message":"Run Script Finished","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00847.html","page":"http://www.toratemetfreeware.com/online/f_00847.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.306Z","context":"behavior","message":"Behaviors finished","details":{"finished":1,"page":"http://www.toratemetfreeware.com/online/f_00847.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.307Z","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"http://www.toratemetfreeware.com/online/f_00847.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.333Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_00848.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.335Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1473,"total":6592,"pending":1,"failed":8,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:27:49.333Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_00848.html\",\"added\":\"2023-09-10T10:07:34.617Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:49.544Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.558Z","context":"behavior","message":"Running behaviors","details":{"frames":1,"frameUrls":["http://www.toratemetfreeware.com/online/f_00848.html"],"page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.558Z","context":"behavior","message":"Run Script Started","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00848.html","page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.562Z","context":"behavior","message":"Run Script Finished","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00848.html","page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.562Z","context":"behavior","message":"Behaviors finished","details":{"finished":1,"page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.563Z","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"http://www.toratemetfreeware.com/online/f_00848.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.590Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_00849.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.592Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1474,"total":6592,"pending":1,"failed":8,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:27:52.589Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_00849.html\",\"added\":\"2023-09-10T10:07:34.617Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:52.769Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.360Z","context":"behavior","message":"Running behaviors","details":{"frames":1,"frameUrls":["http://www.toratemetfreeware.com/online/f_00849.html"],"page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.360Z","context":"behavior","message":"Run Script Started","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00849.html","page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.365Z","context":"behavior","message":"Run Script Finished","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_00849.html","page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.366Z","context":"behavior","message":"Behaviors finished","details":{"finished":1,"page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.367Z","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"http://www.toratemetfreeware.com/online/f_00849.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.542Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_01323.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.543Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1475,"total":6592,"pending":1,"failed":8,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:27:55.393Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_01323.html\",\"added\":\"2023-09-10T10:07:34.618Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:27:55.857Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.804Z","context":"behavior","message":"Running behaviors","details":{"frames":1,"frameUrls":["http://www.toratemetfreeware.com/online/f_01323.html"],"page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.805Z","context":"behavior","message":"Run Script Started","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_01323.html","page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.811Z","context":"behavior","message":"Run Script Finished","details":{"frameUrl":"http://www.toratemetfreeware.com/online/f_01323.html","page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.811Z","context":"behavior","message":"Behaviors finished","details":{"finished":1,"page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.813Z","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"http://www.toratemetfreeware.com/online/f_01323.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.840Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_05077.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:01.841Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1476,"total":6592,"pending":1,"failed":8,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:28:01.839Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_05077.html\",\"added\":\"2023-09-10T10:07:34.618Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:28:02.070Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_05077.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:29:32.071Z","context":"general","message":"Page Load Timeout, skipping page","details":{"msg":"Navigation timeout of 90000 ms exceeded","page":"http://www.toratemetfreeware.com/online/f_05077.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:29:32.071Z","context":"worker","message":"Worker Exception","details":{"type":"exception","message":"Navigation timeout of 90000 ms exceeded","stack":"TimeoutError: Navigation timeout of 90000 ms exceeded\n    at Timeout.<anonymous> (file:///app/node_modules/puppeteer-core/lib/esm/puppeteer/util/Deferred.js:24:33)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)","page":"http://www.toratemetfreeware.com/online/f_05077.html","workerid":0}}
{"logLevel":"warn","timestamp":"2023-09-10T11:29:32.072Z","context":"pageStatus","message":"Page Load Failed","details":{"loadState":0,"page":"http://www.toratemetfreeware.com/online/f_05077.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:29:32.099Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_02339.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:29:32.100Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1477,"total":6592,"pending":1,"failed":9,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:29:32.099Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_02339.html\",\"added\":\"2023-09-10T10:07:34.619Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:29:32.369Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_02339.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:31:02.390Z","context":"general","message":"Page Load Timeout, skipping page","details":{"msg":"Navigation timeout of 90000 ms exceeded","page":"http://www.toratemetfreeware.com/online/f_02339.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:31:02.391Z","context":"worker","message":"Worker Exception","details":{"type":"exception","message":"Navigation timeout of 90000 ms exceeded","stack":"TimeoutError: Navigation timeout of 90000 ms exceeded\n    at Timeout.<anonymous> (file:///app/node_modules/puppeteer-core/lib/esm/puppeteer/util/Deferred.js:24:33)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)","page":"http://www.toratemetfreeware.com/online/f_02339.html","workerid":0}}
{"logLevel":"warn","timestamp":"2023-09-10T11:31:02.391Z","context":"pageStatus","message":"Page Load Failed","details":{"loadState":0,"page":"http://www.toratemetfreeware.com/online/f_02339.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:31:02.419Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_02380.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:31:02.420Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1478,"total":6592,"pending":1,"failed":10,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:31:02.418Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_02380.html\",\"added\":\"2023-09-10T10:07:34.619Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:31:02.680Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_02380.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:32:32.698Z","context":"general","message":"Page Load Timeout, skipping page","details":{"msg":"Navigation timeout of 90000 ms exceeded","page":"http://www.toratemetfreeware.com/online/f_02380.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:32:32.699Z","context":"worker","message":"Worker Exception","details":{"type":"exception","message":"Navigation timeout of 90000 ms exceeded","stack":"TimeoutError: Navigation timeout of 90000 ms exceeded\n    at Timeout.<anonymous> (file:///app/node_modules/puppeteer-core/lib/esm/puppeteer/util/Deferred.js:24:33)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)","page":"http://www.toratemetfreeware.com/online/f_02380.html","workerid":0}}
{"logLevel":"warn","timestamp":"2023-09-10T11:32:32.699Z","context":"pageStatus","message":"Page Load Failed","details":{"loadState":0,"page":"http://www.toratemetfreeware.com/online/f_02380.html","workerid":0}}
{"logLevel":"info","timestamp":"2023-09-10T11:32:32.727Z","context":"worker","message":"Starting page","details":{"workerid":0,"page":"http://www.toratemetfreeware.com/online/f_00908.html"}}
{"logLevel":"info","timestamp":"2023-09-10T11:32:32.728Z","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":1479,"total":6592,"pending":1,"failed":11,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2023-09-10T11:32:32.727Z\",\"url\":\"http://www.toratemetfreeware.com/online/f_00908.html\",\"added\":\"2023-09-10T10:07:36.177Z\",\"depth\":3}"]}}
{"logLevel":"info","timestamp":"2023-09-10T11:32:32.910Z","context":"general","message":"Awaiting page load","details":{"page":"http://www.toratemetfreeware.com/online/f_00908.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:34:02.931Z","context":"general","message":"Page Load Timeout, skipping page","details":{"msg":"Navigation timeout of 90000 ms exceeded","page":"http://www.toratemetfreeware.com/online/f_00908.html","workerid":0}}
{"logLevel":"error","timestamp":"2023-09-10T11:34:02.932Z","context":"worker","message":"Worker Exception","details":{"type":"exception","message":"Navigation timeout of 90000 ms exceeded","stack":"TimeoutError: Navigation timeout of 90000 ms exceeded\n    at Timeout.<anonymous> (file:///app/node_modules/puppeteer-core/lib/esm/puppeteer/util/Deferred.js:24:33)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)","page":"http://www.toratemetfreeware.com/online/f_00908.html","workerid":0}}
{"logLevel":"warn","timestamp":"2023-09-10T11:34:02.932Z","context":"pageStatus","message":"Page Load Failed","details":{"loadState":0,"page":"http://www.toratemetfreeware.com/online/f_00908.html","workerid":0}}
```

**Comment by ikreymer** [comment]
Can you post the exact crawl settings used? This is something that's fairly hard to reproduce, but can try..
There's the healthcheck option, which would probably be failing by now given the consecutive page timeouts.
Even without healthcheck, it should probably exit with a failure

**Comment by rgaudin** [comment]
```sh
crawl \
  --failOnFailedSeed \
  --waitUntil load \
  --depth -1 \
  --timeout 90 \
  --behaviors autoplay,autofetch,siteSpecific \
  --behaviorTimeout 90 \
  --sizeLimit 4294967296 \
  --diskUtilization 90 \
  --timeLimit 7200 \
  --url http://www.toratemetfreeware.com/online/ \
  --userAgentSuffix "Youzim.it+contact+zimfarm@kiwix.org" \
  --cwd /output/.tmp73td1zpp \
  --statsFilename /output/crawl.json
```

**Comment by ikreymer** [comment]
One possible issue is we were catching the wrong event, `page.on('crash')` was from playwright, but puppeteer has `page.on('error')`, which deals with page crashes! This is one possible cause for stuck pages, when page crashes but a new page is not created.

**Comment by ikreymer** [comment]
Fixed several issues that could cause this, hopefully fixed in 0.11.1. Please test with 0.11.1 and let us know if something like this is still happening.
